src/super_gradients/training/dataloaders/dataloaders.py

```diff
@@ -81,6 +82,8 @@ def get_data_loader(config_name: str, dataset_cls: object, train: bool, dataset_
 
     dataloader = DataLoader(dataset=dataset, **dataloader_params)
     dataloader.dataloader_params = dataloader_params
+
+    maybe_setup_dataloader_adapter(dataloader=dataloader)
```

Comment:
I agree with you 99%, but it is technically possible that the user would have a custom dataset (not SG at all), register it and want to use the adapter with recipe.

It's not the most probable use case, and we may prefer to not support such option (at least until it's requested). It's usually a good rule of thumb to avoid unnecessary complexity.
But the solution is here and takes 1 line, and that's why I thought maybe we should just support it already.

No hard opinion on that, we can remove if you still think so

Reply:
Okay okay, let's keep it then)